### PR TITLE
Adding bits to install Salt packages built using Tiamat.

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Adding bits to install Salt packages built using Tiamat.

### What issues does this PR fix or reference?
N/A

### CODE REVIEW

File: bootstrap-salt.sh [hunk 1]
```diff
@@ -7712,6 +8225,143 @@ install_macosx_restart_daemons() {
...
+    tempfile="$(mktemp /tmp/${salt_pkg_name}-XXXXXXXX.tar.gz 2>/dev/null)"
+
+    if [ -z "$tempfile" ]; then
+        echoerror "Failed to create temporary file in /tmp"
+        return 1
+    fi
+
+    __fetch_url "$tempfile" "$salt_pkg_url" || return 1
+
+    #rm -f "$tempfile"
```
Comment (Ch3LL) [inline bootstrap-salt.sh]: does this need to be removed?

**Review by Ch3LL — CHANGES_REQUESTED** [review]
one small question. Also it looks like there are test failures

**Review by Ch3LL — APPROVED** [review]

**Review by krionbsd — APPROVED** [review]

**Comment by myii** [comment]
@garethgreenaway Tested this out in the Formulas' `salt-image-builder` CI (actually used #1845).  We've already been building Docker images for `tiamat` for some time now, so converted things over for `onedir`.  Hit some errors, particularly with RedHat 7 distros.

https://gitlab.com/myii/salt-image-builder/-/pipelines/600150567

* Ignore Debian 9 and 12, since they're not available for `onedir`
* Also ignore CentOS Stream 8, which has some temporary repo `404` errors

#### Amazon Linux 2

```console
*  INFO: Running version: 2022.05.19
 *  INFO: Executed by: 
 *  INFO: Command line: 'sh -XUdfPD -x python3 onedir_rc latest'
 *  WARN: Running the unstable version of bootstrap-salt.sh
 *  INFO: System Information:
 *  INFO:   CPU:          GenuineIntel
 *  INFO:   CPU Arch:     x86_64
 *  INFO:   OS Name:      Linux
 *  INFO:   OS Version:   5.4.109+
 *  INFO:   Distribution: Amazon Linux AMI 2
 * DEBUG: Binaries will be searched using the following $PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
 *  INFO: Installing minion
 *  INFO: Daemons will not be started
100  317k  100  317k    0     0   750k      0 --:--:-- --:--:-- --:--:--  749k
 * DEBUG: install_amazon_linux_ami_2_onedir_deps not found....
 *  INFO: Found function install_amazon_linux_ami_2_deps
 * DEBUG: DEPS_INSTALL_FUNC=install_amazon_linux_ami_2_deps
 * DEBUG: config_amazon_linux_ami_2_onedir_salt not found....
 * DEBUG: config_amazon_linux_ami_2_salt not found....
 * DEBUG: config_amazon_linux_ami_onedir_salt not found....
 * DEBUG: config_amazon_linux_ami_salt not found....
 *  INFO: Found function config_salt
 * DEBUG: CONFIG_SALT_FUNC=config_salt
 * DEBUG: preseed_amazon_linux_ami_2_onedir_master not found....
 * DEBUG: preseed_amazon_linux_ami_2_master not found....
 * DEBUG: preseed_amazon_linux_ami_onedir_master not found....
 * DEBUG: preseed_amazon_linux_ami_master not found....
 *  INFO: Found function preseed_master
 * DEBUG: PRESEED_MASTER_FUNC=preseed_master
 * DEBUG: INSTALL_FUNC_NAMES=install_amazon_linux_ami_2_onedir install_amazon_linux_ami_2_onedir install_amazon_linux_ami_onedir
 * DEBUG: install_amazon_linux_ami_2_onedir not found....
 * DEBUG: install_amazon_linux_ami_onedir not found....
 * DEBUG: INSTALL_FUNC=null
 * DEBUG: install_amazon_linux_ami_2_onedir_post not found....
 * DEBUG: install_amazon_linux_ami_2_post not found....
 * DEBUG: install_amazon_linux_ami_onedir_post not found....
 * DEBUG: install_amazon_linux_ami_post not found....
 * DEBUG: POST_INSTALL_FUNC=null
 * DEBUG: install_amazon_linux_ami_2_onedir_restart_daemons not found....
 *  INFO: Found function install_amazon_linux_ami_2_restart_daemons
 * DEBUG: STARTDAEMONS_INSTALL_FUNC=install_amazon_linux_ami_2_restart_daemons
 * DEBUG: daemons_running_amazon_linux_ami_2_onedir not found....
 * DEBUG: daemons_running_amazon_linux_ami_2 not found....
 * DEBUG: daemons_running_amazon_linux_ami_onedir not found....
 * DEBUG: daemons_running_amazon_linux_ami not found....
 *  INFO: Found function daemons_running_onedir
 * DEBUG: DAEMONS_RUNNING_FUNC=daemons_running_onedir
 * DEBUG: CHECK_SERVICES_FUNC=null
 * ERROR: No installation function found. Exiting...
```

* The `install_amazon_linux_ami_onedir*` functions haven't been added to the bootstrap script.

#### CentOS 7 & Oracle Linux 7

```console
 * ERROR: https://repo.saltproject.io/salt_rc/salt/py3/redhat/7/x86_64/base/RPM-GPG-KEY-CentOS-7 failed to download to /tmp/salt-gpg-FwdrQQEM.pub
 * ERROR: Failed to run install_centos_onedir_deps()!!!
```

```console
 * ERROR: https://repo.saltproject.io/salt_rc/salt/py3/redhat/7/x86_64/base/RPM-GPG-KEY-CentOS-7 failed to download to /tmp/salt-gpg-MC6OK3N3.pub
 * ERROR: Failed to run install_oracle_linux_onedir_deps()!!!
```

* https://repo.saltproject.io/salt_rc/salt/py3/redhat/7/x86_64/base doesn't exist.

**Comment by garethgreenaway** [comment]
@myii thanks!  Will check these issues out.

**Pull request merged**